app/controllers/assignments_controller.rb

```diff
@@ -22,14 +22,17 @@ def show
 
   def start
     authorize @assignment
-    @project = current_user.projects.new
-    @project.name = current_user.name + "/" + @assignment.name
-    @project.assignment_id = @assignment.id
-    @project.project_access_type="Private"
-    @project.save
-    redirect_to user_project_path(current_user,@project)
-
-
+    if Project.find_by(author_id: current_user.id, assignment_id: @assignment.id).nil?
+      @project = current_user.projects.new
+      @project.name = current_user.name + "/" + @assignment.name
+      @project.assignment_id = @assignment.id
+      @project.project_access_type="Private"
+      @project.save
+      redirect_to user_project_path(current_user,@project)
+    else
+      @project = Project.find_by(author_id: current_user.id, assignment_id: @assignment.id)
```

Comment:
This code will become redundant once the initial assignment page is closed because [this](https://github.com/CircuitVerse/CircuitVerse/blob/master/app/views/groups/show.html.erb#L149) if condition will never allow "Start Working" to be ever printed again. 
In a way, choosing the "Start Working" option once the project has already been initialized should "not authorized". 
What do you guys think?

Reply:
Yeah, the issue though assumes if someone opens the next page in a new tab( like most of @phyzicist 's students have been doing according to him. )